### PR TITLE
Added test + implementation for current relations

### PR DESCRIPTION
**Pull request by tfrancart**

### CODE REVIEW

File: ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt [hunk 1]
```diff
@@ -26,6 +26,8 @@
 			<label>Relation hiérarchique</label>
 			<shortcutIfSubjectIsSourceOfRelation>rico:hasOrHadSubordinate</shortcutIfSubjectIsSourceOfRelation>
 			<shortcutIfSubjectIsTargetOfRelation>rico:isOrWasSubordinateTo</shortcutIfSubjectIsTargetOfRelation>
+			<shortcutIfSubjectIsSourceOfRelation_current>rico:hasDirectSubordinate</shortcutIfSubjectIsSourceOfRelation_current>
```
Comment (florenceclavaud) [inline ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt:29]: Yes, and there should also be something like shortcutIfSubjectIsSourceOfRelation_past with value rico:hadSubordinate (and the second one, with value rico:wasSubordinateTo)

File: ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt [hunk 2]
```diff
@@ -46,6 +50,8 @@
 			<label>Relation de contrôle</label>
 			<shortcutIfSubjectIsSourceOfRelation>rico:isOrWasControllerOf</shortcutIfSubjectIsSourceOfRelation>
 			<shortcutIfSubjectIsTargetOfRelation>rico:hasOrHadController</shortcutIfSubjectIsTargetOfRelation>
+			<shortcutIfSubjectIsSourceOfRelation_current>rico:isOrWasControllerOf</shortcutIfSubjectIsSourceOfRelation_current>
```
Comment (florenceclavaud) [inline ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt:53]: Nothing should change here. We do not have wasControllerOf or isControllerOf in RiC-O for now

File: ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt
Comment (florenceclavaud) [inline ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico-relations.xslt:1]: IMHO a few details should be changed in this file; see my specific comments

File: ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico.xslt [hunk 3]
```diff
@@ -767,11 +767,23 @@
...
-        		<xsl:with-param name="uri" select="eac2rico:URI-AgentExternal(@xlink:href, document(concat($INPUT_FOLDER, '/', @xlink:href, '.xml')))"/>	
-        	</xsl:call-template>
-        </xsl:element>
+		<xsl:variable name="shortcutProperty">
+			<xsl:choose>
+				<xsl:when test="not(eac:dateRange/eac:toDate)">
+					<xsl:value-of select="$HIERARCHICAL_RELATION_CONFIG/*[local-name() = normalize-space($type)]/shortcutIfSubjectIsSourceOfRelation_current" />
+				</xsl:when>
+				<xsl:otherwise>
+					<xsl:value-of select="$HIERARCHICAL_RELATION_CONFIG/*[local-name() = normalize-space($type)]/shortcutIfSubjectIsSourceOfRelation" />
```
Comment (florenceclavaud) [inline ricoconverter/ricoconverter-convert/src/main/resources/xslt_eac/eac2rico.xslt:776]: or `<xsl:value-of select="$HIERARCHICAL_RELATION_CONFIG/*[local-name() = normalize-space($type)]/shortcutIfSubjectIsSourceOfRelation_past" />`? See my comment on the eac2rico-relations.xslt

**Review by florenceclavaud — COMMENTED** [review]
I have added a few comments to the changes in the XSLT files. IMHO a few lines are missing (to take into account the cases of past relations, which correspond to other object properties, that are not exactly shortcuts but were added to RiC-O 1.0). Also, we should probably add other unit tests, and change some existing ones (I can help do this). Finally, even if in the AnF, such relations _always_ have a date, there may be cases elsewhere where they are not dated, and we could take them into account too - and generate what is in fact already generated woth Ric-O Converter 2.

**Comment by tfrancart** [comment]
- Si date de début mais pas de date de fin, utiliser la relation "current"
- Si date de début + date de fin, utiliser une relation "past"
- Si aucune date, rester sur la propriété générique

**Comment by tfrancart** [comment]
@florenceclavaud I did the "past" relation implementation. Can you please review and also review the unit tests, and add unit tests if you feel some cases of relations with dates are missing ? thanks

File: ricoconverter/ricoconverter-convert/src/test/resources/eac2rico/_37emptyDates/expected.xml [hunk 4]
```diff
@@ -34,7 +34,7 @@
          </rico:AgentName>
       </rico:hasOrHadAgentName>
       <rico:agentIsSourceOfAgentHierarchicalRelation rdf:resource="agentHierarchicalRelation/005205-111111-19470101-*"/>
-      <rico:hasOrHadSubordinate rdf:resource="agent/111111" />
+      <rico:hadSubordinate rdf:resource="agent/111111" />
```
Comment (florenceclavaud) [inline ricoconverter/ricoconverter-convert/src/test/resources/eac2rico/_37emptyDates/expected.xml:37]: It should be rico:hasDirectSubordinate here (in the input file there is no end date for the relation, so we consider it is current).

**Review by florenceclavaud — CHANGES_REQUESTED** [review]
Everything is Ok except in the last file (expected file in the _37emptyDates folder) - see my comment. And so, I am also realizing that there may be eac:toDate or eac:fromDate that are empty; the XSLT does not take this into acccount I think.

**Comment by florenceclavaud** [comment]
> * Si date de début mais pas de date de fin, utiliser la relation "current"
> 
>     * Si date de début + date de fin, utiliser une relation "past"
> 
>     * Si aucune date, rester sur la propriété générique

Ou plutôt : si date de début existe et est non vide, mais pas de date de fin ou date de fin vide, on utilise la relation "current"
Si date de début et date de fin non vides, on utilise la relation "past"
Si pas de dates du tout, ou si fromDate et toDate sont vides, on reste sur la propriété générique

**Comment by tfrancart** [comment]
Note : to determine that the relation is in the past, we could even avoid testing on the presence of the fromDate attribute. The only presence of the `toDate` attribute is sufficient.

**Review by florenceclavaud — APPROVED** [review]
Everything OK

**Pull request merged**